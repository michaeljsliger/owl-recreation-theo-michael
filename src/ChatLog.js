import React, {Component} from 'react';
import ChatMessage from './ChatMessage';
import chatEvents from './ChatSTORE';
// import STORE from './STORE'

// in array of JSX elements, 
// match id to STORE id for obj
// if type message {
    // display message
    // timestamp new Date(props.time)
//}


class ChatLog extends Component {
// takes in store
    render() {
        let key = 0;
        const chatMessageArray = chatEvents.map(el => {
                
                return (<li className="chat-message-box" key={key++}><ChatMessage type={el.type} data-id={el.participantId} message={el.message} time={el.time}/></li>)
        })


        return (
        <ul className="chat-list">{chatMessageArray}</ul>
        )
    }
}

export default ChatLog;