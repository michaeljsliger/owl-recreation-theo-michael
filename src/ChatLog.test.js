import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import ChatLog from './ChatLog';
import renderer from 'react-test-renderer';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('Chat Log Branch Testing', () => {
  // smoke test

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatLog />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  // snapshot test
  it('should render correctly', () => {
    const tree = renderer
          .create(<ChatLog />)
          .toJSON();
    expect(tree).toMatchSnapshot();
  })
})