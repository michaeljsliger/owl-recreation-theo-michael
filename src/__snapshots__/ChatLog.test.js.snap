// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Log Branch Testing should render correctly 1`] = `
<ul
  className="chat-list"
>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      <div
        className="single-message-header"
      >
        <div>
          <h5>
            Axe Kubicka
          </h5>
        </div>
        <div>
          <h6>
            1/30/2019, 5:50:46 AM
          </h6>
        </div>
      </div>
      <div
        className="time-container"
      >
        <p>
          Hello world
        </p>
      </div>
    </div>
  </li>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      Caty Flucker
       
      gave a thumbs up!
    </div>
  </li>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      Frank Runciman
       
      gave a thumbs down!
    </div>
  </li>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      Koren Templeton
       
      raised their hand!
    </div>
  </li>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      Koren Templeton
       
      clapped
    </div>
  </li>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      Koren Templeton
       
      joined
    </div>
  </li>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      Ashla Attwool
       
      left
    </div>
  </li>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      Axe Kubicka
       
      joined the stage
    </div>
  </li>
  <li
    className="chat-message-box"
  >
    <div
      className="single-message"
    >
      Axe Kubicka
       
      leaved the stage
    </div>
  </li>
</ul>
`;
